scala/scala.bzl

```diff
@@ -72,11 +73,30 @@ def _compile(ctx, jars, buildijar):
       ijar=ctx.file._ijar.path,
       out=ctx.outputs.jar.path,
       ijar_out=ctx.outputs.ijar.path)
+  #TODO here is the issue...when scala library depends on this, it's not picking up anything
+
+  #TODO I need a command that will explode out the source jars
+  #TODO probably can be in a helper function
+  srcjar_cmd = "\n_SRCJAR_ARG=\" \""
+  srcjars = _srcjar_filetype.filter(ctx.files.srcjars + list(_collect_srcjars(ctx.attr.deps)))
+  if len(srcjars) > 0:
+    # TODO Somehow, the manifest is sneaking in here?! so we filter... but really
+    #      should try and understand why the manifest is being put here!
+    # TODO the FileType.filter also seemed to get rid of it...??
+    for srcjar in srcjars:
+      srcjar_cmd += """
+mkdir -p {{out}}_tmp/expand_srcjars/{srcjar}
+unzip {srcjar} -d {{out}}_tmp/expand_srcjars/{srcjar}
```

Comment:
can you comment about the double escaping here? It is pretty tricky at first glance.